### PR TITLE
Stack transition validation

### PR DESCRIPTION
**Pull request by mahaling**
validation of stacks during transition. This mainly checks for any missing tiles especially during apply mipmaps and apply lens correction. 
A generalized validation has been added to perform this check wherever necessary.

### CODE REVIEW

File: rendermodules/module/render_module.py [hunk 1]
```diff
@@ -109,6 +109,19 @@ def output_tilespecs_to_stack(self, tilespecs, output_stack=None,
             output_stack, tilespecs, sharedTransforms=sharedTransforms,
             pool_size=pool_size, close_stack=close_stack, render=render)
 
+    def validate_tilespecs(self, input_stack, output_stack, z, render=None):
```
Comment (djkapner) [inline rendermodules/module/render_module.py]: I suggest calling this as optional from this function:
```
def output_tilespecs_to_stack(...validate=False...):
    if validate:
        self.validate_tilespecs(self.input_stack, self.output_stack)
```

File: rendermodules/module/render_module.py [hunk 2]
```diff
@@ -109,6 +109,19 @@ def output_tilespecs_to_stack(self, tilespecs, output_stack=None,
...
+        missing_ts = []
+
+        render = self.render if render is None else render
+
+        input_rs = renderapi.resolvedtiles.get_resolved_tiles_from_z(input_stack, z, render=render)
+        output_rs = renderapi.resolvedtiles.get_resolved_tiles_from_z(output_stack, z, render=render)
+        if len(input_rs.tilespecs) != len(output_rs.tilespecs): # pragma: no cover
```
Comment (djkapner) [inline rendermodules/module/render_module.py]: maybe instead of checking just len() check tileids in a set overlap way

Comment (djkapner) [reply rendermodules/module/render_module.py]: this is what I meant here:
```
in_ids = set([t.tileId for t in input_rs.tilespecs])                                                                               
out_ids = set([t.tileId for t in output_rs.tilespecs])                                                                             
if in_ids != out_ids:
    etc.
```

File: rendermodules/module/render_module.py (hunk 1 above)
Comment (djkapner) [reply rendermodules/module/render_module.py]: we punted on this after discussion

File: rendermodules/module/render_module.py (hunk 2 above)
Comment (mahaling) [reply rendermodules/module/render_module.py]: ok... have added the changes

**Comment by RussTorres** [comment]
What is the point of the "recent changes" commit?  I don't see any code differences, but it looks like a lot of files are entirely overwritten by it.  Why is this, and can we scrub it for the sake of commit history?

**Comment by mahaling** [comment]
> What is the point of the "recent changes" commit? I don't see any code differences, but it looks like a lot of files are entirely overwritten by it. Why is this, and can we scrub it for the sake of commit history?

could be an effect of merge. I do not recall the changes made in that commit. If it helps, please scrub it.

**Pull request merged**